### DISCUSSION TITLE
Conditional side effect

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#156] [Q&A]
Is this conditional side effect execution bad or problematic? It seems to work fine, but I remember that you mentioned it is a bad practice. 

```dart
wlib.WarmUpVersionableData<VotePersonal>? votePersonalCapsule(
  CapsuleHandle use,
) {
  final info = use(_getVotingInfoCapsule);

  if (info == null) return null;

  return use.warmUpVersionableData(
    'Vote personal',
    use.memo(
      () => info.then((info) {
        debug('Personal: ${info.voteInfo}');
        return info.voteInfo;
      }),
      [info],
    ),
  );
}
```

In this case, as it uses either `zero` or `one` side effect, and never could happen to mistaken two side effects with different type, I suppose that is the reason to be working okay. But still, do you consider it a bad practice?

**Comment by busslina** [comment]
I workaround this by using `use.nullableFuture` instead `nullableFuture` inside `use.warmUpVersionableData`. Now it looks like this:
```dart
wlib.WarmUpVersionableData<VotePersonal>? votePersonalCapsule(
  CapsuleHandle use,
) {
  final info = use(_getVotingInfoCapsule);

  return use.warmUpVersionableData(
    'Vote personal',
    use.memo(
      () => info?.then((info) => info.voteInfo),
      [info],
    ),
  );
}
```

**Reply by GregoryConrad** [reply]
This is fine, but I’d move the use.memo up above the warmUpVersionableData call. Just helps with readability a bit.

definitely don’t conditionally call an effect like you were in the original comment, that is prone to break in a future version. Even if it works now, that is not guaranteed behavior
